### ISSUE TITLE
[C++][Parquet] Loading big parquet files leads to huge memory consumption

### ISSUE THREAD

**Issue by KuczaRacza** [apache/arrow#44890]
### Describe the bug, including details regarding any error messages, version, and platform.

Loading big parquet files (23GB file) leads to high memory consumption, ending in OOM. Memory consumption surpass my machine limit of 64GB ram.

Reader is set up similar as  demonstrated on docs page via `` parquet::arrow::FileReaderBuilder`` with key difference of
 passing custom allocator. Huge usage of memory is not report by my custom allocator nor by memory pool passed to reader.  

After invastigation iI encouter thath RangeReadeCache created lot of  cache entries, 735 to be exact witch each about 33MiB of length. Each with it's future with use count 1. 
![enteries](https://github.com/user-attachments/assets/e2eb65ab-8163-4bfd-b4cd-fc3dd2335f11)
Also for some reason this uses default memory pool
![image](https://github.com/user-attachments/assets/95b78844-e590-4e67-8c89-45c0d440516b)

### Component(s)

C++

**Comment by assignUser** [comment]
Is the parquet file using compression?

**Comment by KuczaRacza** [comment]
Yes it uses compression, although it shouldn't take so much memory, it's read by batches limited to 8192

**Comment by assignUser** [comment]
It's of course hard to say without a minimal reproducible example but it sounds like your inadvertently loading the entire file into memory (735*33 ~ 23G) and is then decompressed which results in OOM as 23G compressed can easily surpass 64G uncompressed.

**Comment by KuczaRacza** [comment]
here is my minimal example of that behavior [https://gist.github.com/KuczaRacza/4ddcae7d39aef7bfbe352017b96e15f9](example). I was wrong about 20+GiB file  swelling to 60+ GiB in RAM other components in my workflow contributed to this high usage. But it look like reading by batches takes roughly same amount of memory as size of the file. It's too much to handle any files bigger than  few GiBs

**Comment by mapleFU** [comment]
> After invastigation iI encouter thath RangeReadeCache created lot of cache entries, 735 to be exact witch each about 33MiB of length. Each with it's future with use count 1.

Sorry for late reply.

By default, `RangeReadCache` will try to cache all file you need, it assume you're reading the object storage and need optimization.

You can disable it and using buffer reader, which creates a buffer for each column just in one row-group to avoid OOM

**Comment by KuczaRacza** [comment]
Thanks for help, looks like removing ``  reader_properties.enable_buffered_stream();``  and  setting  `` arrow_reader_props.set_pre_buffer(false);`` helps.  Setting  `` arrow_reader_props.set_pre_buffer(false);`` solely ends up with abort.  Perhaps documentation should warn users about this potential OOM.  Thanks again fro help with resolving this.
